code/modules/mob/living/carbon/superior_animal/golem/golem.dm

```diff
@@ -106,3 +115,89 @@ GLOBAL_LIST_INIT(golems_special, list(/mob/living/carbon/superior_animal/golem/s
 		var/obj/structure/obstacle = locate(/obj/structure) in T
 		if(obstacle)
 			obstacle.attack_generic(src, rand(surrounds_mult * melee_damage_lower, surrounds_mult * melee_damage_upper), pick(attacktext), TRUE)
+
+/mob/living/carbon/superior_animal/golem/loseTarget()
+	GLOB.active_golems -= src
+	. = ..()
```

Comment:
I don't seem to find a moment when anything is added to `active_golems` list, let alone when golem acquires a new target.